lib/news_scraper/configuration.rb

```diff
@@ -0,0 +1,16 @@
+module NewsScraper
+  class Configuration
+    DEFAULT_SCRAPE_PATTERNS_FILEPATH = File.expand_path('../../../config/article_scrape_patterns.yml', __FILE__)
+    attr_accessor :scrape_patterns, :scrape_patterns_filepath
```

Comment:
I don't like how `scrape_patterns` can also be set without regard to the `scrape_patterns_filepath`.

You could end up with a `scrape_patterns` that isn't the same as the scrape patterns in `scrape_patterns_filepath`.

Only one of these should be writable too, and that's probably `scrape_patterns_filepath`.